core/src/main/java/io/micronaut/core/convert/DefaultConversionService.java

```diff
@@ -924,6 +957,16 @@ protected void registerDefaultConverters() {
                 new MultiValuesConverterFactory.MapToMultiValuesConverter(this));
         addConverter(Object.class, io.micronaut.core.convert.value.ConvertibleMultiValues.class,
                 new MultiValuesConverterFactory.ObjectToMultiValuesConverter(this));
+
+        Collection<TypeConverterRegistrar> registrars = new ArrayList<>();
+        ForkJoinPool forkJoinPool = new ForkJoinPool(); // Create a custom ForkJoinPool to prevent deadlock
```

Comment:
could also use `SoftServiceLoader.disableFork` instead

Reply:
this would also remove the need for the SoftServiceLoader changes below (though it could impact performance)

Reply:
I'm afraid it's going to be the first touch of the jars filesystem and it will cause a slowdown.

Reply:
should definitely measure this, creating an entirely new fjp will have overhead from starting all the threads, too